xarray/core/datatree.py

```diff
@@ -447,14 +447,10 @@ def __init__(
         --------
         DataTree.from_dict
         """
-        if children is None:
-            children = {}
-
-        super().__init__(name=name)
         self._set_node_data(_to_new_dataset(dataset))
 
-        # shallow copy to avoid modifying arguments in-place (see GH issue #9196)
-        self.children = {name: child.copy() for name, child in children.items()}
+        # comes after setting node data as this will check for clashes between child names and existing variable names
+        super().__init__(name=name, children=children)
```

Comment:
If you try to swap these back around you get an error from trying to access `._node_coordinates` before it has been set. I think this part of the code could be refactored a little (in a follow-up PR). Preferably we would set the data using a codepath that checks for name collisions between variables and children, instead of relying on the assignment of children to do that step.